src/app.js

```diff
@@ -0,0 +1,98 @@
+const express = require("express");
+const app = express();
+const { User, Matcher, Party, UserParty } = require("../models/index")
+const { db } = require("../db/connection");
+const { check, validationResult } = require('express-validator')
+const { Op } = require('@sequelize/core')
+
+app.use(express.json());
+app.use(express.urlencoded());
+
+// User routes (mainly for manual testing)
+app.get("/users", async(req, res) => {
+    const allUsers = await User.findAll();
+    res.send(allUsers);
+});
+
+app.get("/users/:id", async(req, res) => {
+    const id = req.params.id;
+    const user = await User.findByPk(id);
+    res.send(user);
+});
+
+app.post("/users", async (req, res) => {
+    const createdUser = await User.create(req.body);
+    res.json(createdUser);
+});
+
+app.delete("/users/:id", async (req, res) => {
+    const deleted = await User.destroy({
+        where: {
+            id: req.params.id
+        }
+    });
+    res.sendStatus(200);
+});
+
+
+// Matcher Routes
+app.post("/users/:id/matcher", async (req, res) => {
+    const createdMatcher = await Matcher.create(req.body);
+    await User.findByPk(req.params.id)
+        .then(user => user.addMatcher(createdMatcher));
+    res.json(createdMatcher);
+});
+
+app.delete("/users/:userId/matcher/:matcherId", async (req, res) => {
+    await Matcher.destroy({
+        where: {
+            id: req.params.matcherId,
+            userId: req.params.userId
+        }
+    });
+    res.sendStatus(200);
+});
+
+app.put("/users/:userId/matcher/:matcherId", async (req, res) => {
+    const matcher = await Matcher.findOne({
+        where: {
+            id: req.params.matcherId,
+            userId: req.params.userId
+        }
+    });
+    const updatedMatcher = await matcher.update(req.body);
+    res.json(updatedMatcher);
+});
+
+// Match Route
+app.get("/users/:userId/matcher/:matcherId/match", async (req, res) => {
+    const user1 = await User.findByPk(req.params.userId);
+    const userMatcher = await Matcher.findByPk(req.params.matcherId);
+    const match = await Matcher.findOne({
+        where: {
+            userId: { [Op.ne]: userMatcher.UserId},
+            platform: userMatcher.platform,
+            gameName: userMatcher.gameName
+        }
+    });
+    const user2 = await User.findByPk(match.dataValues.UserId);
+    const newParty = await Party.create({
```

Comment:
If this is how we're going to create a party and add users to it, do we still need the UserParty Model? This is how I did the associations and I didn't create or need that through table. As of right now it looks like the only place it is being used is the import? (I could be wrong github doesn't have the same hover and highlight like Code Search does but when I ctr + f and search UserParty I only get the import)